framework/include/problems/FEProblemBase.h

```diff
@@ -1713,6 +1777,34 @@ FEProblemBase::finalizeUserObjects(const MooseObjectWarehouse<T> & warehouse)
 
       if (pp)
         _pps_data.storeValue(pp->PPName(), pp->getValue());
+
+      auto vpp = std::dynamic_pointer_cast<VectorPostprocessor>(object);
+
+      if (vpp)
+        vpps_finalized.insert(vpp->PPName());
+    }
+
+    // Broadcast/Scatter any VPPs that need it
+    for (auto & vpp_name : vpps_finalized)
+    {
+      auto & vpp_vectors = _vpps_data.vectorPostprocessorVectors(vpp_name);
+
+      for (auto & current_pair : vpp_vectors._values)
+      {
+        auto & vpp_state = current_pair.second;
+
+        if (!vpp_vectors._is_broadcast && vpp_state.needs_broadcast)
+        {
+          unsigned int size = vpp_state.current->size();
```

Comment:
Just curious. You are explicit with the type here. Does `parallel.h` not work with `std::size_t`?